crates/core/app/src/app_version/component.rs

```diff
@@ -16,7 +16,8 @@ fn version_to_software_version(version: u64) -> &'static str {
         6 => "v0.77.x",
         7 => "v0.79.x",
         8 => "v0.80.x",
-        9 => "v0.81.x",
+        9 => "v1.0.x",
+        10 => "v2.0.x",
```

Comment:
> the transition from `0.81.x` to `1.0.x` maintained compatibility with APP_VERSION 9, and that may present difficulties during the upgrade for nodes on one or the other version.

The clobbering of the `9` match is what I expect will introduce problems if used as-is on mainnet, where different versions are surely running. Must we extend the matching logic to accept multiple-choice? I don't see a way around that, unless we mandate upgrading 0.81.x -> 1.0.x -> 2.0.x.

Reply:
I thought UIP-6 only checked the `APP_VERSION`, can take a look

Reply:
On closer inspection, I agree: the crate version string is only ever used in user-facing messaging, so we should be good to go here.